File: December 2024 Leetcode Solution/December-13.py
class Solution:
    def findScore(self, nums: List[int]) -> int:
        n = len(nums)
        vis = [False] * n
        indexed_nums = sorted((x, i) for i, x in enumerate(nums))  
        ans = 0

        for x, i in indexed_nums:
            if not vis[i]: 
                ans += x
                vis[i] = True
                if i - 1 >= 0:  
                    vis[i - 1] = True
                if i + 1 < n: 
                    vis[i + 1] = True

        return ans

File: December 2024 Leetcode Solution/December-13.rs
impl Solution {
    pub fn find_score(nums: Vec<i32>) -> i64 {
        let n = nums.len();
        let mut indexed_nums: Vec<(i32, usize)> = nums.into_iter().enumerate().map(|(i, x)| (x, i)).collect();
        indexed_nums.sort_unstable();

        let mut vis = vec![false; n];
        let mut ans: i64 = 0;

        for (val, idx) in indexed_nums {
            if !vis[idx] {
                ans += val as i64;
                vis[idx] = true;
                if idx > 0 {
                    vis[idx - 1] = true;
                }
                if idx + 1 < n {
                    vis[idx + 1] = true;
                }
            }
        }

        ans
    }
}
